docs/en/integrations/data-ingestion/kafka/kafka-table-engine.md

```diff
@@ -168,24 +214,22 @@ You should see 200,000 rows:
 
 ##### Stopping & restarting message consumption
 
-To stop message consumption, simply detach the Kafka engine table e.g.
+To stop message consumption, you can detach the Kafka engine table:
 
 ```sql
 DETACH TABLE github_queue;
 ```
 
-Note that this will not impact the offsets of the consumer group. To restart consumption, and continue from the previous offset, simply reattach the table.
+This will not impact the offsets of the consumer group. To restart consumption, and continue from the previous offset, reattach the table.
 
 ```sql
 ATTACH TABLE github_queue;
 ```
 
-We can use this operation to make setting and schema changes - see below.
-
 ##### Adding Kafka Metadata
 
-
-It is not uncommon for users to need to identify the coordinates of the original Kafka messages for the rows in ClickHouse. For example, we may want to know how much of a specific topic or partition we have consumed. For this purpose, the Kafka table engine exposes several [virtual columns](../../../engines/table-engines/index.md#table_engines-virtual_columns). These can be persisted as columns in our target table by modifying our schema and materialized view’s select statement.
+It can be useful to keep track of the metadata from the original Kafka messages after it's been ingested into ClickHouse.
+For example, we may want to know how much of a specific topic or partition we have consumed. For this purpose, the Kafka table engine exposes several [virtual columns](../../../engines/table-engines/index.md#table_engines-virtual_columns). These can be persisted as columns in our target table by modifying our schema and materialized view’s select statement.
```

Comment:
```suggestion
It can be useful to keep track of the metadata from the original Kafka messages after it's been ingested into ClickHouse. For example, we may want to know how much of a specific topic or partition we have consumed. For this purpose, the Kafka table engine exposes several [virtual columns](../../../engines/table-engines/index.md#table_engines-virtual_columns). These can be persisted as columns in our target table by modifying our schema and materialized view’s select statement.
```